src/parser/sparqlParser/SparqlQleverVisitor.h

```diff
@@ -368,20 +373,10 @@ class SparqlQleverVisitor : public SparqlAutomaticVisitor {
 
   antlrcpp::Any visitFunctionCall(
       SparqlAutomaticParser::FunctionCallContext* ctx) override {
-    // TODO<Hannah>: duplicate code in visitIriOrFunction.
-    auto& iri = visitIri(ctx->iri()).as<std::string>();
-    auto& argList =
-        visitArgList(ctx->argList()).as<std::vector<ExpressionPtr>>();
-    if (iri == "<http://www.opengis.net/def/function/geosparql/distance>") {
-      if (argList.size() != 2) {
-        throw SparqlParseException{
-            "Function geof:distance requires two arguments"};
-      }
-      return createExpression<sparqlExpression::DistExpression>(
-          std::move(argList[0]), std::move(argList[1]));
-    } else {
-      throw SparqlParseException{"IRI function \"" + iri + "\" not supported"};
-    }
+    return processIriFunctionCall(
+        std::move(visitIri(ctx->iri()).as<std::string>()),
```

Comment:
If you have the const string& , you can drop the first `move` (from the string). I assume that for getting rid of the second move we would need to patch antlr (respectively wait for a new release which uses std::any instead of antlrcpp::Any, which then will have all the required overloads.

Reply:
Done!